src/test/blackbox.rs

```diff
@@ -5,36 +5,45 @@ use crate::{
 	types,
 };
 use jsonrpc_core_client::{transports::local, RpcChannel};
+use crate::node::TestRuntimeRequirements;
 
-pub enum BlackBoxNode<Runtime> {
+/// A black box node, either runs a manual seal background node,
+/// or connects via ws to a running node.
+pub enum BlackBoxNode<N> {
 	/// Connects to an external node.
 	External(String),
 	/// Spawns a pristine node.
-	Internal(InternalNode<Runtime>),
+	Internal(InternalNode<N>),
 }
 
 /// A black box test.
-pub struct BlackBox<Runtime> {
-	node: BlackBoxNode<Runtime>,
+pub struct BlackBox<N> {
+	node: BlackBoxNode<N>,
 }
 
-impl<Runtime> BlackBox<Runtime>
-where
-	Runtime: frame_system::Trait,
+impl<N> BlackBox<N>
+	where
+		N: TestRuntimeRequirements,
 {
-	pub async fn with_state<R>(&mut self, closure: impl FnOnce() -> R) -> R {
-		TestExternalities::<Runtime>::new(self.rpc()).execute_with(closure)
+	/// Execute provided `Fn` in an externalities provided environment.
+	pub async fn with_state<T>(&mut self, func: impl FnOnce() -> T) -> T {
+		TestExternalities::<N>::new(self.rpc()).execute_with(func)
+	}
+	
+	/// Wait `number` of blocks.
+	pub fn wait_blocks(&self, _number: impl Into<types::BlockNumber<N::Runtime>>) {
+		// no-op
```

Comment:
Let's leave it a s a TODO maybe? The implementation should take current block from RPC and then wait until the `_number` is actually reached.